lib/codec.js

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {termCodecs} from './codecs/codecMap';
+import {Base64PadCodec} from './codecs/base64Pad';
+import {CborldError} from './error';
+import {CodecMapCodec} from './codecs/codecMap';
+import {UrlCodec} from './codecs/url';
+import {ContextCodec} from './codecs/context';
+import {SimpleTypeCodec} from './codecs/simpleType';
+import {XsdDateTimeCodec} from './codecs/xsdDateTime';
+
+/**
+ * Gets a list of all term codecs available to use when transforming term
+ * values.
+ *
+ * @returns {Array} - An array of all term codecs.
+ */
+export function getTermCodecs() {
+  const codecs = [];
+  termCodecs.forEach((value, key) => {
+    if(typeof key === 'string') {
+      codecs.push(key);
+    }
+  });
+
+  return codecs;
+}
+
+/**
+ * Gets a compression map for the given set of JSON-LD Context URLs.
+ *
+ * @param {object} [args] - The arguments to the function.
+ * @param {Array} [args.contextUrls] - The JSON-LD Context URLs to use when
+ *   building the compression map.
+ * @param {object} [args.appTermMap] - A map of JSON-LD terms and
+ *   their associated CBOR-LD compression codecs.
+ * @param {object} [args.documentLoader] - The document loader to use when
+ *   dereferencing URLs.
+ * @param {object} [args.encode] - If true, a map of term encoding codecs are
+ *   returned.
+ * @param {object} [args.decode] - If true, a map of term decoding codecs are
+ *   returned.
+ *
+ * @returns {object} - The compression map.
+ */
+export async function getTermCodecMap({
+  contextUrls, appTermMap, documentLoader, encode, decode}) {
+  const termCodecMap = new Map();
+  appTermMap = appTermMap || new Map();
+
+  // retrieve all JSON-LD Contexts
+  const contexts = await Promise.all(contextUrls.map(
+    async contextUrl => await documentLoader(contextUrl)));
+
+  // generate a compression map for each context
+  contexts.forEach(context => {
+    const map = _generateTermEncodingMap(context);
+    map.forEach((value, key) => {
+      termCodecMap.set(key, value);
+    });
```

Comment:
Return values from `forEach` are thrown away, so this can be shortened.

```suggestion
    map.forEach((value, key) => termCodecMap.set(key, value));
```